hail/src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala

```diff
@@ -3,93 +3,83 @@ package is.hail.expr.ir.functions
 import is.hail.asm4s
 import is.hail.asm4s._
 import is.hail.expr.ir._
-import is.hail.expr.types.physical.{PBoolean, PCanonicalString, PInt32, PType}
+import is.hail.expr.types.physical.{PBoolean, PCanonicalString, PInt32, PLocus, PType}
 import is.hail.expr.types.virtual._
 import is.hail.utils._
 import is.hail.variant.ReferenceGenome
 
 class ReferenceGenomeFunctions(rg: ReferenceGenome) extends RegistryFunctions {
+  def rgCode(mb: EmitMethodBuilder[_], rg: ReferenceGenome): Code[ReferenceGenome] = mb.getReferenceGenome(rg)
+  var registered: Set[(String, Type, Seq[Type], Seq[Type])] = Set()
```

Comment:
Investigating whether we still want to be able to ReferenceGenome.clear(), in HailContext.stop(), now that we are correctly parameterizing ReferenceGenome registered functions. 

If we don't want that, I can get rid of all registerRGCode uses below, as well as the related code in ReferenceGenome.